tests/by-util/test_ls.rs

```diff
@@ -6311,3 +6311,220 @@ fn test_ls_time_sort_without_long() {
+
+    // With -f, files should NOT be sorted
+    let unsorted = scene.ucmd().arg("-f").succeeds().stdout_move_str();
+    assert!(unsorted.contains("zebra"));
+    assert!(unsorted.contains("apple"));
+    assert!(unsorted.contains("banana"));
+}
+
+#[test]
+fn test_f_flag_disables_implicit_color() {
+    // Test that -f disables implicit color (not explicitly set)
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    at.mkdir("dir");
+    at.touch("file.txt");
+
+    // -f without explicit --color: should not have color
+    let result = scene.ucmd().arg("-f").succeeds().stdout_move_str();
+    assert!(
+        !result.contains("\x1b["),
+        "Color should be disabled with -f alone"
+    );
+}
+
+#[test]
+fn test_explicit_color_always_works_with_f() {
+    // Test that explicit --color always enables color, regardless of -f
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    at.mkdir("dir");
+    at.touch("file.txt");
+
+    // --color then -f: explicit --color should still enable color
+    let result1 = scene
+        .ucmd()
+        .arg("--color=always")
+        .arg("-f")
+        .succeeds()
+        .stdout_move_str();
+    assert!(
+        result1.contains("\x1b["),
+        "Explicit --color should work even with -f after"
+    );
+
+    // -f then --color: explicit --color should enable color
+    let result2 = scene
+        .ucmd()
+        .arg("-f")
+        .arg("--color=always")
+        .succeeds()
+        .stdout_move_str();
+    assert!(
+        result2.contains("\x1b["),
+        "Explicit --color should work even with -f before"
+    );
+}
+
+#[test]
+fn test_f_overrides_a_and_big_a() {
+    // Test last-flag-wins: -f after -a/-A
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    at.touch("visible");
+    at.touch(".hidden");
+
+    // -A then -f: should show all files
+    scene
+        .ucmd()
+        .arg("-A")
+        .arg("-f")
+        .succeeds()
+        .stdout_contains(".hidden");
+
+    // -f then -A: should honor -A (almost all)
+    let output = scene
+        .ucmd()
+        .arg("-f")
+        .arg("-A")
+        .succeeds()
+        .stdout_move_str();
+    assert!(output.contains(".hidden"));
+}
+
+#[test]
+fn test_f_overrides_sort_flags() {
+    // Test last-flag-wins: -f overrides -t, -S, -v, -X
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    at.touch("file1.txt");
+    at.touch("file2.txt");
+    at.touch("file3.txt");
+
+    // -t then -f: should disable sorting
+    let unsorted = scene
+        .ucmd()
+        .arg("-t")
+        .arg("-f")
+        .succeeds()
+        .stdout_move_str();
+    assert!(unsorted.contains("file1.txt"));
+    assert!(unsorted.contains("file2.txt"));
+    assert!(unsorted.contains("file3.txt"));
+
+    // -f then -t: should honor -t (sort by time)
+    scene.ucmd().arg("-f").arg("-t").succeeds();
```

Comment:
Both scenes don't ensure that the last flag wins. In the first case you only ensure the files are shown, and in the second that there is no error.